py/examples/graphics_spline.py

```diff
@@ -17,47 +18,59 @@
 
 splines = [
     # Lines
-    g.spline(x=x, y=y, **line_style),  # same as curve='linear'
-    g.spline(x=x, y=y, curve='basis', **line_style),
-    g.spline(x=x, y=y, curve='basis-closed', **line_style),
-    g.spline(x=x, y=y, curve='basis-open', **line_style),
-    g.spline(x=x, y=y, curve='cardinal', **line_style),
-    g.spline(x=x, y=y, curve='cardinal-closed', **line_style),
-    g.spline(x=x, y=y, curve='cardinal-open', **line_style),
-    g.spline(x=x, y=y, curve='smooth', **line_style),
-    g.spline(x=x, y=y, curve='smooth-closed', **line_style),
-    g.spline(x=x, y=y, curve='smooth-open', **line_style),
-    g.spline(x=x, y=y, curve='linear', **line_style),
-    g.spline(x=x, y=y, curve='linear-closed', **line_style),
-    g.spline(x=x, y=y, curve='monotone-x', **line_style),
-    g.spline(x=x, y=y, curve='monotone-y', **line_style),
-    g.spline(x=x, y=y, curve='natural', **line_style),
-    g.spline(x=x, y=y, curve='step', **line_style),
-    g.spline(x=x, y=y, curve='step-after', **line_style),
-    g.spline(x=x, y=y, curve='step-before', **line_style),
+    g.spline(x=[float(v) for v in x], y=[float(v) for v in y], curve='linear', style=line_style),
+    g.spline(x=[float(v) for v in x], y=[float(v) for v in y], curve='basis', style=line_style),
+    g.spline(x=[float(v) for v in x], y=[float(v) for v in y], curve='basis-closed', style=line_style),
```

Comment:
This makes code harder to follow. Is there any other way to make mypy happy while preserving the original way of passing params to spline?

Reply:
Updated with a function to make it clear, I've try to modify x outside of the list and it seems not working